questionpy_server/decorators.py

```diff
@@ -39,10 +39,9 @@ def decorator(function: RouteHandler) -> RouteHandler:
         require_question_state = question_state_type is bytes
         main_part_json_model: type[MainBaseModel] | None = type_hints.get("data")
 
-        if main_part_json_model:
-            assert issubclass(
-                main_part_json_model, MainBaseModel
-            ), f"Parameter 'data' of function {function.__name__} has unexpected type."
+        if main_part_json_model and not issubclass(main_part_json_model, MainBaseModel):
+            msg = f"Parameter 'data' of function {function.__name__} has unexpected type."
+            raise TypeError(msg)
```

Comment:
wrt <https://docs.astral.sh/ruff/rules/f-string-in-exception/>

Hier den Code weniger leserlich zu machen, damit der Stacktrace minimal klarer ist, sehe ich nicht ganz. Ist keine starke Meinung, aber ich würde die Regel lieber global ausmachen.

Reply:
Wird der Code weniger leserlich? Finde entschachtelte Instruktionen tendenziell eher förderlich für die Lesbarkeit, z.B.

```py
msg = f"'{InitWorker.__name__}' message expected, '{type(init_msg).__name__}' received"
raise WorkerNotInitializedError(msg)

# anstatt

raise WorkerNotInitializedError(
    f"'{InitWorker.__name__}' message expected, '{type(init_msg).__name__}' received"
)
```

Aber ich hab da auch keine starke Meinung zu. Mein Ansatz war soweit sinnvoll alle Regeln auf default/recommended zu belassen und von da aus auszudünnen.